src/System.Windows.Forms/src/System/Windows/Forms/DragDropHelper.cs

```diff
@@ -0,0 +1,601 @@
+                    ref NativeMethods.ActiveX.IID_IUnknown,
+                    out object obj).ThrowIfFailed();
+
+                Debug.WriteLineIf(CompModSwitches.DragDrop.TraceInfo, "TryGetDropTargetHelper IDropTargetHelper created");
+                dropTargetHelper = (IDropTargetHelper)obj;
+                return true;
+            }
+            catch (COMException ex)
+            {
+                Debug.WriteLineIf(CompModSwitches.DragDrop.TraceInfo, $"TryGetDropTargetHelper COMException {ex}");
+                dropTargetHelper = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        ///  Notifies the drag-image manager that the drop target has been entered, and provides it with the information
+        ///  needed to display the drag image.
+        /// </summary>
+        public static void DragEnter(IntPtr hwndTarget, IComDataObject dataObject, ref Point ppt, uint dwEffect)
+        {
+            if (!TryGetDropTargetHelper(out IDropTargetHelper? dropTargetHelper))
+            {
+                return;
+            }
+
+            try
+            {
+                dropTargetHelper.DragEnter(hwndTarget, dataObject, ref ppt, dwEffect).ThrowIfFailed();
+            }
+            catch (COMException ex)
+            {
+                Debug.WriteLineIf(CompModSwitches.DragDrop.TraceInfo, $"IDropTargetHelper::DragEnter COM error {ex}");
+                return;
+            }
+            finally
+            {
+                Marshal.FinalReleaseComObject(dropTargetHelper);
+            }
+
+            SetIsNewDragImage(dataObject, false);
+        }
+
+        /// <summary>
+        ///  Notifies the drag-image manager that the cursor position has changed and provides it with the information
+        ///  needed to display the drag image.
+        /// </summary>
+        public static void DragOver(IntPtr hwndTarget, IComDataObject dataObject, ref Point ppt, uint dwEffect)
+        {
+            // If the application has set a new drag image, e.g. in DropSource.GiveFeedback, we must call DragEnter
+            // before calling DragOver in order for the Windows drag image manager to effectively display the drag image.
+            if (GetIsNewDragImage(dataObject))
+            {
+                DragEnter(hwndTarget, dataObject, ref ppt, dwEffect);
+            }
+
+            if (!TryGetDropTargetHelper(out IDropTargetHelper? dropTargetHelper))
+            {
+                return;
+            }
+
+            try
+            {
+                dropTargetHelper.DragOver(ref ppt, dwEffect).ThrowIfFailed();
+            }
+            catch (COMException ex)
+            {
+                Debug.WriteLineIf(CompModSwitches.DragDrop.TraceInfo, $"IDropTargetHelper::DragOver COM error {ex}");
+                return;
+            }
+            finally
+            {
+                Marshal.FinalReleaseComObject(dropTargetHelper);
```

Comment:
Do not use `Marshal.FinalReleaseComObject` where you can use simply `Marshal.ReleaseComObject`, When conversion to ComWrappers happens it would be much easier and mechanical change. 
```suggestion
                Marshal.ReleaseComObject(dropTargetHelper);
```